src/components/api/fetchArticles.js

```diff
@@ -1,7 +1,8 @@
-function fetchArticles() {
-	return fetch('https://nc-news-a2p9.onrender.com/api/articles').then((response) =>
-		response.json()
-	);
+function fetchApi(endpoint) {
+	const baseUrl = 'https://nc-news-a2p9.onrender.com/api/';
+	return fetch(baseUrl + endpoint).then((response) => response.json());
 }
 
-export default fetchArticles;
+export default function fetchArticles() {
+	return fetchApi('articles');
```

Comment:
I see your reasoning here but you don't need to do this. Take a look at the example here: https://notes.northcoders.com/courses/js-front-end/axios#axios-instance